src/HttpHeader.cc

```diff
@@ -486,6 +493,17 @@ HttpHeader::parse(const char *header_start, size_t hdrLen, Http::ContentLengthIn
             return 0;
         }
 
+        if (lines > 1 || hasBareCr) {
+            const auto framingHeader = (e->id == Http::HdrType::CONTENT_LENGTH || e->id == Http::HdrType::TRANSFER_ENCODING);
+            if (framingHeader) {
+                debugs(55, warnOnError, "WARNING: " << (hasBareCr ? hasBareCr : "obs-fold") << " seen in framing-sensitive " << e->name << ": " << e->value);
```

Comment:
Forgot to mention: We have already warned about bare CR. I do not insist on reducing this WARNING noise in this PR, but it is easy to reduce if you want to:

```suggestion
                if (!hasBareCr) // already warned about bare CRs
                    debugs(55, warnOnError, "WARNING: " << "obs-fold in framing-sensitive " << e->name << ": " << e->value);
```

A better version would warn about both obs-fold and bare CR if both were found, but you have rejected similar improvements earlier in this PR, so I am not going to revise them here.